plugins/modules/idrac_reset.py

```diff
@@ -92,42 +167,253 @@
   }
 '''
 
-
+import re
 import json
-from ansible_collections.dellemc.openmanage.plugins.module_utils.dellemc_idrac import iDRACConnection, idrac_auth_params
+import time
+import socket
+import subprocess
+from urllib.error import HTTPError, URLError
+from copy import deepcopy
+from ansible_collections.dellemc.openmanage.plugins.module_utils.idrac_redfish import iDRACRedfishAPI, idrac_auth_params
 from ansible.module_utils.basic import AnsibleModule
-from ansible.module_utils.six.moves.urllib.error import URLError, HTTPError
+from ansible.module_utils.compat.version import LooseVersion
 from ansible.module_utils.urls import ConnectionError, SSLValidationError
+from ansible_collections.dellemc.openmanage.plugins.module_utils.utils import (
+    get_idrac_firmware_version, wait_after_idrac_reset)
 
 
-def run_idrac_reset(idrac, module):
-    if module.check_mode:
-        msg = {'Status': 'Success', 'Message': 'Changes found to commit!', 'changes_applicable': True}
-    else:
-        idrac.use_redfish = True
-        msg = idrac.config_mgr.reset_idrac()
-    return msg
+SYSTEMS_URI = "/redfish/v1/Systems"
+MANAGERS_URI = "/redfish/v1/Managers"
+IDRAC_RESET_RETRIES = 10
+LC_STATUS_CHECK_SLEEP = 30
+IDRAC_RESET_OPTIONS_URI = "/redfish/v1/Managers/iDRAC.Embedded.1"
+iDRAC_JOB_URI = "/redfish/v1/Managers/iDRAC.Embedded.1/Jobs/{job_id}"
+IDRAC_RESET_LIFECYCLE_STATUS_URI =  "/redfish/v1/Managers/iDRAC.Embedded.1/Oem/Dell/DellLCService/Actions/DellLCService.GetRemoteServicesAPIStatus"
```

Comment:
Check if possible, make URI fetching dynamic, don't hardcode the values

Reply:
Updated!